File: src/domain/entities/order.ts
import type { Client } from "./client";
import type { PaymentMethod } from "./payment_method";

// Order interface definition
export interface Order {
	id_order: number;
	order_date: Date;
	status: string;
	total_amount: number;
	delivery_type: string;
	id_client: number;
	id_cart: number;
	id_payment_method: number;
	client?: Client;
	paymentMethod: PaymentMethod;
}

File: backend/realtime_service/internal/models/order.go
package models

import "time"

// Order representa un pedido simplificado para eventos en tiempo real.
// Sincronizado con OrderEntity del rest_service.
type Order struct {
	IDOrder         int       `json:"id_order"`
	OrderDate       time.Time `json:"order_date"`
	Status          string    `json:"status"` // "pending", "confirmed", "delivered", etc.
	TotalAmount     float64   `json:"total_amount"`
	DeliveryType    string    `json:"delivery_type"` // "pickup", "delivery"
	IDClient        int       `json:"id_client"`
	IDCart          int       `json:"id_cart"`
	IDPaymentMethod int       `json:"id_payment_method"`
	IDDelivery      *int      `json:"id_delivery,omitempty"` // Opcional (nullable)
}
